src/o365/spo/commands/folder/folder-move.ts

```diff
@@ -0,0 +1,340 @@
+    Promise<void> {
+
+    let pollCount: number = 0;
+    let retryAttemptsCount: number = 0;
+
+    const checkCondition = (resolve: () => void, reject: (error: any) => void): void => {
+      pollCount++;
+      const requestUrl: string = `${opts.webUrl}/_api/site/GetCopyJobProgress`;
+      const requestOptions: any = {
+        url: requestUrl,
+        headers: Utils.getRequestHeaders({
+          authorization: `Bearer ${opts.accessToken}`,
+          'accept': 'application/json;odata=nometadata'
+        }),
+        body: { "copyJobInfo": opts.copyJopInfo },
+        json: true
+      };
+
+      if (this.debug) {
+        cmd.log('getCopyJobProgress request...');
+        cmd.log(requestOptions);
+      }
+
+      request.post(requestOptions).then((resp: any): void => {
+        retryAttemptsCount = 0; // clear retry on promise success 
+
+        if (this.debug) {
+          cmd.log('getCopyJobProgress response...');
+          cmd.log(resp);
+        }
+
+        if (this.verbose) {
+          if (resp.JobState && resp.JobState === 4) {
+            cmd.log(`Check #${pollCount}. Copy job in progress... JobState: ${resp.JobState}`);
+          }
+          else {
+            cmd.log(`Check #${pollCount}. JobState: ${resp.JobState}`);
+          }
+        }
+
+        for (const item of resp.Logs) {
+          const log = JSON.parse(item);
+
+          // reject if progress error 
+          if (log.Event === "JobError" || log.Event === "JobFatalError") {
+            return reject(log.Message);
+          }
+        }
+
+        // three possible scenarios
+        // job done = success promise returned
+        // job in progress = recursive call using setTimeout returned
+        // max poll attempts flag raised = reject promise returned
+        if (resp.JobState === 0) {
+          // job done
+          resolve();
+        }
+        else if (pollCount < opts.progressMaxPollAttempts) {
+          // if the condition isn't met but the timeout hasn't elapsed, go again
+          setTimeout(checkCondition, opts.progressPollInterval, resolve, reject);
+        }
+        else {
+          reject(new Error('getCopyJobProgress timed out'));
+        }
+      },
+        (error: any) => {
+          retryAttemptsCount++;
+
+          // let's retry x times in row before we give up since
+          // this is progress check and even if rejects a promise
+          // the actual move process can success.
+          if (retryAttemptsCount <= opts.progressRetryAttempts) {
+            setTimeout(checkCondition, opts.progressPollInterval, resolve, reject);
+          }
+          else {
+            reject(error);
+          }
+        });
+    };
+
+    return new Promise<void>(checkCondition);
+  }
+
+  /**
+   * Combines base and relative url 
+   * considering any missing slashes
+   * @param baseUrl https://contoso.com
+   * @param relativeUrl sites/abc
+   */
+  private urlCombine(baseUrl: string, relativeUrl: string): string {
```

Comment:
Why not use the standard `url.resolve` function instead?

Reply:
@waldekmastykarz  I used this since it is in the `spo folder copy`.

Correct me if I'm wrong, the url.resolve does combine and make a url of it, just "as it is"

This method `urlCombine` handles if there´s any slashes at the end and removes them.

So I guess we have to make a decision here what to use. What do you think?

Reply:
According to the docs it should handle it correctly: https://nodejs.org/api/url.html#url_url_resolve_from_to. Have you seen otherwise?

Reply:
So it depends of if we should care about "fixing" the last ´/`
Depending on what you think regarding this, I think we should make the same changes to `spo file copy` and `spo folder copy` later :)

Reply:
Agreed that `spo file copy` and `spo folder copy` should work the same way. Is there any particular scenario where we explicitly care about the trailing `/`?

Reply:
@waldekmastykarz I actually dont know. Maybe @phawrylak knows?
My guess, is that we could use the url.resolve and just change the unit tests slightly.